### DISCUSSION TITLE
Use @OneOf with Date

### DISCUSSION THREAD

**Discussion by janek515** [TypedProject/ts-express-decorators#2206] [Q&A] [answered]
How would you come about using Date with the OneOf decorator? No matter what I tried I couldn't get it to produce the correct schema.
```ts
    @OneOf(Date, OtherType)
    @Optional()
    property?: Date | OtherType;
```
The example above doesn't work, it just treats Date as an object. How can you create a `JsonSchema` for a date?

**Comment by Romakita** [comment, marked as answer]
It’s always possible to set custom raw jsonschema Using a plain object js or using schema builder:

```ts
import {date, OneOf) from "@tsed/schema";
@OneOf(date(), OtherType)
```

Any decorator should also works `@Any(Date, Type)`.

see you
Romain
